asset/api/Asset.Samples.Tests/ListAssetsTest.cs

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using Xunit;
+
+[Collection(nameof(AssetFixture))]
+public class ListAssetsTest
+{
+    private readonly AssetFixture _fixture;
+    private readonly ListAssetsSample _sample;
+
+    public ListAssetsTest(AssetFixture fixture)
+    {
+        _fixture = fixture;
+        _sample = new ListAssetsSample();
+    }
+
+    [Fact(Skip = "https://github.com/GoogleCloudPlatform/dotnet-docs-samples/issues/1141")]
+    public void TestListAssets()
+    {
+        // Run the sample code.
+        var result = _sample.ListAssets(_fixture.ProjectId);
+
+        string assetName = String.Format("//storage.googleapis.com/{0}", _fixture.BucketName);
+        Assert.Contains(assetName, result.ToString());

```

Comment:
This will fail if the asset is not on the first page of results. You need to use Assert.Contains

Reply:
I've added an additional parameter assetTypes to restrict the result to GCS bucket only, I guess this should avoid the issue?

Reply:
I know that some of the other Asset tests are checking against the string representation of the responses and I've flagged that on review before, it's really not accurate. You should navigate the response until you find what you are looking for, something like this:

```
Assert.Contains(result, asset => asset.Name == assetName);
```
In this way you guarante that results are fetched until we find the one we are looking for, and also, that if by chance, there's another property for some other asset containing the value `assetName` we don't get a false positive.

Reply:
Thanks. I was not familiar with C#. Glad that C# has this way to get what we need.